### PR TITLE
75 auth window change

### PR DESCRIPTION
**Pull request by pavelparfishov**
<html>
<body>
<h2 dir="auto" style="box-sizing: border-box; margin-top: 0px !important; margin-bottom: 16px; font-size: 1.5em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--color-border-muted); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Task merge request</h2>

Related Issue | [Изменить окно авторизации#75](https://github.com/SUAI-TaskPlanner-Contest/TaskPlanner/issues/75)
-- | --
Квалификация ревьюера | Знание требований заказчика

Исполнители: @pavelparfishov

<h3 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"></h3><h3 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a id="user-content-code-changes" class="anchor" aria-hidden="true" href="https://github.com/SUAI-TaskPlanner-Contest/TaskPlanner/blob/main/Workflow%20Guides/Templates/MrMessage.md#code-changes" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; float: left; padding-right: 4px; margin-left: -20px; line-height: 1;"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg>

</a>Code changes</h3> Был изменен QML-код для окна авторизации: добавлены новые шрифты, изменены цвета и дизайн.  Еще были добавлены обработчики нажатия на кнопки и изменение цвета кнопки при нажатии для визуального удобства.

<h3 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a id="user-content-ui-demonstration" class="anchor" aria-hidden="true" href="https://github.com/SUAI-TaskPlanner-Contest/TaskPlanner/blob/main/Workflow%20Guides/Templates/MrMessage.md#ui-demonstration" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; float: left; padding-right: 4px; margin-left: -20px; line-height: 1;"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg>
</a>UI Demonstration</h3>

image

<h3 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 0px !important; font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a id="user-content-how-to-test" class="anchor" aria-hidden="true" href="https://github.com/SUAI-TaskPlanner-Contest/TaskPlanner/blob/main/Workflow%20Guides/Templates/MrMessage.md#how-to-test" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none; float: left; padding-right: 4px; margin-left: -20px; line-height: 1;"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg>
</a>How to test</h3> <p>Проверить наличие необходимых полей и визуальную привлекательность окна. Также проверить корректность конструкций на Python</p>

</body>
</html>

### CODE REVIEW

File: Code/Auth_window/main.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            self._errorText = value
+            self.errorTextChanged.emit(value)
+
+    @pyqtSlot()
+    def button1Clicked(self):
+        print("Button 1 clicked")
+
+    @pyqtSlot()
+    def localareaClicked(self):
+        print("Local area clicked")
```
Comment (astronik00) [inline Code/Auth_window/main.py:30]: Такие же замечания, как в #66
Убрать print, оставить pass

File: Code/Auth_window/window.qml [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+// Основное окно
+ApplicationWindow {
+    visible: true
+    id: rectangle
+    width: 400
+    height: 530
+    color: "transparent"
+    title: "Окно авторизации"
+    
+    flags: Qt.FramelessWindowHint // Установите флаг для отображения окна без системной строки
```
Comment (astronik00) [inline Code/Auth_window/window.qml:14]: этого комментария не должно быть в мейне

File: Code/Auth_window/window.qml [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        y: 369
+        width: 250
+        height: 39
+        text: qsTr("Войти")
+       
+        //font.capitalization: Font.Capitalize
+        checkable: true
+        //font.pointSize: 14
+        //display: AbstractButton.TextOnly
+        palette.buttonText: "white"
```
Comment (astronik00) [inline Code/Auth_window/window.qml:195]: если код не используется, то лучше его убрать, либо пометить, зачем он может быть нужен

Comment (pavelparfishov) [reply Code/Auth_window/window.qml]: @astronik00 Вроде все лишнее везде убрал. Добавил файл с примерами использования [Readme.md](https://github.com/SUAI-TaskPlanner-Contest/TaskPlanner/blob/75-auth-window-change/Code/Auth_window/Readme.md)

**Comment by pavelparfishov** [comment]
@astronik00  изменил окно после ознакомления с окном Ярослава

image

File: Code/Auth_window/main.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            self._errorText = value
+            self.errorTextChanged.emit(value)
+
+    @pyqtSlot()
+    def localareaClicked(self):
+        pass
+
+    @pyqtSlot(str, str, str, str)
+    def loginClicked(self, login, password, calendarAddress, serverLink):
+        pass
```
Comment (astronik00) [inline Code/Auth_window/main.py:30]: Также этот класс нужно вынести в отдельный файл. И это не ButtonHandler, а тогда AuthWindowHandler, ибо обрабатывает он все окно, а не конкретно кнопки. В файле main.py находится только вызов твоего окна авторизации, в общем какие-то глобальные вещи. Все детали реализации должны быть вынесены.

File: Code/Auth_window/main.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+
+if __name__ == "__main__":
+    app = QApplication(sys.argv)
+    engine = QQmlApplicationEngine()
+
+    buttonHandler = ButtonHandler()
+    engine.rootContext().setContextProperty("buttonHandler", buttonHandler)
```
Comment (astronik00) [inline Code/Auth_window/main.py:38]: эти 2 строки можно в 1 сократить, ибо больше объект нигде не используется

**Review by astronik00 — CHANGES_REQUESTED** [review]
В целом все хорошо, вот мои последние замечания. Также попробуй сделать squash, чтобы все было теперь одним коммитом. Если не получается - пиши, я помогу.

File: Code/Auth_window/main.py (hunk 4 above)
Comment (pavelparfishov) [reply Code/Auth_window/main.py]: @astronik00 Вынес класс в отдельный файл. Сделал squash. Вроде получилось

**Comment by pavelparfishov** [comment]
@astronik00 удалил лишнюю папку с AuthWindow. Убрал синее свечение при наведении на кнопку. Проверил вывод окна с помощью кода из инструкции - работает
image

**Pull request merged**